src/Agent.Plugins/PipelineCache/FingerprintCreator.cs

```diff
@@ -98,6 +98,39 @@ internal struct Enumeration
             public SearchOption Depth;
         }
 
+        internal class MatchedFile
+        {
+            public MatchedFile(string displayPath, long fileLength, string hash)
+            {
+                this.DisplayPath = displayPath;
+                this.FileLength = fileLength;
+                this.Hash = hash;    
+            }
+
+            public string DisplayPath;
+            public long FileLength;
+            public string Hash;
+
+            public string GetHash() {
+                return MatchedFile.GenerateHash(new [] { this });
+            }
+
+            public static string GenerateHash(IEnumerable<MatchedFile> matches) {                
+                string s = matches.Aggregate(new StringBuilder(),
+                        (sb, file) => sb.Append($"\nSHA256({file.DisplayPath})=[{file.FileLength}]{file.Hash}"),
+                        sb => sb.ToString());
+
+                var sha256 = new SHA256Managed();
```

Comment:
>new SHA256Managed(); [](start = 29, length = 20)

pretty sure we can just keep a singleton for this, static variable in MatchedFile?

Reply:
Good suggestion. I moved this, and also was able to get rid of the sha256 object in `EvaluateKeyToFingerprint`.